rotkehlchen/tests/unit/test_evm_tx_decoding.py

```diff
@@ -201,3 +205,66 @@ def test_query_and_decode_transactions_works_with_different_chains(
     assert len(hashes) == 0
     hashes = dbevmtx.get_transaction_hashes_not_decoded(chain_id=ChainID.ETHEREUM, limit=None, addresses=None)  # noqa: E501
     assert len(hashes) == 1
+
+
+@pytest.mark.parametrize('ethereum_accounts', [[
+    '0x756F45E3FA69347A9A973A725E3C98bC4db0b5a0',
+    '0x9328D55ccb3FCe531f199382339f0E576ee840A3',
+    '0x4bba290826c253bd854121346c370a9886d1bc26',
+]])
+def test_genesis_remove_address(
+        database: 'DBHandler',
+        ethereum_accounts: list[ChecksumEvmAddress],
+        ethereum_transaction_decoder: 'EthereumTransactionDecoder',
+):
+    """
+    Checks that if an address had a genesis transacion:
+    1. The decoded event gets deleted when the address is removed
+    2. Genesis tx gets removed if it was the last tracked address with a genesis tx
+    """
+    dbevmtx = DBEvmTx(database)
+    dbevents = DBHistoryEvents(database)
+    genesis_address_1, genesis_address_2, no_genesis_address = ethereum_accounts
+
+    def get_events() -> list[HistoryBaseEntry]:
+        with database.conn.read_ctx() as cursor:
+            events = dbevents.get_history_events(
+                cursor=cursor,
+                filter_query=HistoryEventFilterQuery.make(event_identifiers=[GENESIS_HASH]),
+                has_premium=True,
+            )
+
+        return events
+
+    def delete_address(address: ChecksumEvmAddress) -> None:
```

Comment:
You are not deleting the address but the transactions related to this address. The name is confusing

Reply:
Yes. Change the name here .
```suggestion
    def delete_transactions_for_address(address: ChecksumEvmAddress) -> None:
```